eclass/libretro-core.eclass

```diff
@@ -0,0 +1,197 @@
+
+case "${EAPI:-0}" in
+	6|7)
+		EXPORT_FUNCTIONS src_unpack src_prepare src_compile src_install
+		;;
+	*)
+		die "EAPI=${EAPI} is not supported" ;;
+esac
+
+# @FUNCTION: libretro-core_src_unpack
+# @DESCRIPTION:
+# The libretro-core src_unpack function which is exported.
+#
+# This function retrieves the remote Libretro core info files.
+libretro-core_src_unpack() {
+	# If this is a live ebuild, retrieve this core's remote repository.
+	if [[ ${PV} == *9999 ]]; then
+		git-r3_src_unpack
+		# Add used commit SHA for version information, the above could also work.
+		LIBRETRO_COMMIT_SHA=$(git -C "${WORKDIR}/${P}" rev-parse HEAD)
+	# Else, unpack this core's local tarball.
+	else
+		default_src_unpack
+	fi
+}
+
+# @FUNCTION: libretro-core_src_prepare
+# @DESCRIPTION:
+# The libretro-core src_prepare function which is exported.
+#
+# This function prepares the source by making custom modifications.
+libretro-core_src_prepare() {
+	ebegin "Attempting to hack Makefiles to use custom cflags"
+	# Populate COMMIT for GIT_VERSION
+	CUSTOM_LIBRETRO_COMMIT_SHA="\" ${LIBRETRO_COMMIT_SHA:0:7}\""
+	local makefile
+	local flags_modified=0
+	local shopt_saved=$(shopt -p nullglob)
+	shopt -s nullglob
+	for makefile in "${S}"/[Mm]akefile* "${S}"/target-libretro/[Mm]akefile*; do
+		# * Convert CRLF to LF
+		# * Expand *FLAGS to prevent potential self-references
+		# * Where LDFLAGS directly define the link version
+		#   script append LDFLAGS and LIBS
+		# * Where SHARED is used to provide shared linking
+		#   flags ensure final link command includes LDFLAGS
+		#   and LIBS
+		# * Always use $(CFLAGS) when calling $(CC)
+		# * Add short-rev to Makefile
+		[[ -e "${makefile}" ]] && sed \
+			-e 's/\r$//g' \
+			-e "/flags.*=/s:-O[[:digit:]]:${CFLAGS}:g" \
+			-e "/CFLAGS.*=/s:-O[[:digit:]]:${CFLAGS}:g" \
+			-e "/.*,--version-script=.*/s:$: ${LDFLAGS} ${LIBS}:g" \
+			-e "/\$(CC)/s:\(\$(SHARED)\):\1 ${LDFLAGS} ${LIBS}:" \
+			-e 's:\(\$(CC)\):\1 \$(CFLAGS):g' \
+			-e "s/GIT_VERSION\s.=.*$/GIT_VERSION=${CUSTOM_LIBRETRO_COMMIT_SHA}/g" \
+			-i "${makefile}" || die "Failed to use custom cflags in ${makefile}"
+	done
+	${shopt_saved}
+	eend
+	export OPTFLAGS="${CFLAGS}"
+	default_src_prepare
+}
+
+# @VARIABLE: myemakeargs
+# @DEFAULT_UNSET
+# @DESCRIPTION:
+# Optional emake arguments as a bash array. Should be defined before calling
+# src_compile.
+# @CODE
+# src_compile() {
+#       local myemakeargs=(
+#		$(usex neon "HAVE_NEON=1" "")
+#	)
+#       libretro-core_src_configure
+# }
+# @CODE
+
+# @FUNCTION: libretro-core_src_compile
+# @DESCRIPTION:
+# The libretro-core src_compile function which is exported.
+#
+# This function compiles the shared library for this Libretro core.
+libretro-core_src_compile() {
+	# most (if not all) libretro makefiles use DEBUG=1
+	# to enable additional debug features.
+	emake CC=$(tc-getCC) CXX=$(tc-getCXX) \
+		$(usex debug "DEBUG=1" "") "${myemakeargs[@]}" \
+		$([[ -f makefile.libretro ]] && echo '-f makefile.libretro') \
```

Comment:
At most one should exist; I haven't been able to come up with a decent one-liner that does that. I tried:
```
        emake CC=$(tc-getCC) CXX=$(tc-getCXX) \
                $(usex debug "DEBUG=1" "") "${myemakeargs[@]}" \
                -f [Mm]akefile.libretro
```
But that doesn't work if `[Mm]akefile.libretro` doesn't exist. I also tried:
```
$([[ -f [Mm]akefile.libretro ]] && echo '-f [Mm]akefile.libretro')
```
but that doesn't work because there is no file name expansion inside `[[ ]]``

So, I think the two lines I originally had are the most readable solution.

Reply:
I suppose you could do

    $([[ -f Makefile.libretro || -f makefile.libretro ]] && echo -f [Mm]akefile.libretro)

but I agree that's probably less readable.